pkg/controller/lifecycle/actuator.go

```diff
@@ -130,29 +133,56 @@ func (a *actuator) Reconcile(ctx context.Context, ex *extensionsv1alpha1.Extensi
 		return err
 	}
 
-	genericKubeconfigName := extensions.GenericTokenKubeconfigSecretNameFromCluster(cluster)
-	seedResources, err := getSeedResources(oidcReplicas, hibernated, namespace, genericKubeconfigName, oidcShootAccessSecret)
+	// initialize SecretsManager based on Cluster object
+	var (
+		caName        = "ca-extension-shoot-oidc-service"
+		secretConfigs = []extensionssecretsmanager.SecretConfigWithOptions{
+			{
+				Config: &secretutils.CertificateSecretConfig{
+					Name:       caName,
+					CommonName: caName,
+					CertType:   secretutils.CACert,
+				},
+				Options: []secretsmanager.GenerateOption{secretsmanager.Persist()},
+			},
+			{
+				Config: &secretutils.CertificateSecretConfig{
+					Name:       constants.WebhookTLSecretName,
+					CommonName: constants.ApplicationName,
+					DNSNames:   kutil.DNSNamesForService(constants.ApplicationName, namespace),
+					CertType:   secretutils.ServerCert,
+				},
+				// use current CA for signing server cert to prevent mismatches when dropping the old CA from the webhook
+				// config in phase Completing
+				Options: []secretsmanager.GenerateOption{secretsmanager.SignedByCA(caName, secretsmanager.UseCurrentCA)},
+			},
+		}
+	)
+
+	secretsManager, err := extensionssecretsmanager.SecretsManagerForCluster(ctx, a.logger.WithName("secretsmanager"), clock.RealClock{}, a.client, cluster, constants.SecretsManagerIdentity, nil)
```

Comment:
You need to pass the `secretConfigs` in order for the CA to be rotated.
```suggestion
	secretsManager, err := extensionssecretsmanager.SecretsManagerForCluster(ctx, a.logger.WithName("secretsmanager"), clock.RealClock{}, a.client, cluster, constants.SecretsManagerIdentity, secretConfigs)
```

Reply:
nice catch, thank you!